core/src/com/fourinachamber/fortyfive/map/detailMap/SeededMapGenerator.kt

```diff
@@ -104,12 +114,36 @@ class SeededMapGenerator(
                 val choice = allWeightEnds.indexOf(allWeightEnds.first { it >= rndMy })
                 val (_, eventCreator, nodeTexture) = restrictions.optionalEvents[choice]
                 curNode.event = eventCreator()
+                val event = curNode.event
+                if (event is EncounterMapEvent) event.encounterModifierNames = getModifier()
                 curNode.nodeTexture = nodeTexture
             }
             nodesWithoutEvents.remove(curNode)
         }
     }
 
+    private fun getModifier(): Set<String> {
+        val encounterProps = restrictions.encounterProps.toMutableList()
+        val res: MutableSet<String> = mutableSetOf()
+        val nbrOfEncounterIndex =
+            RandomCardSelection.getRandomIndex(restrictions.nbrOfEncounters.map { it.second.toFloat() }
+                .toMutableList(), rnd)
+        repeat(restrictions.nbrOfEncounters[max(nbrOfEncounterIndex, 0)].first) {
+            if (encounterProps.isEmpty()) return res
+            val encIndex = RandomCardSelection.getRandomIndex(encounterProps.map { it.second.toFloat() }
+                .toMutableList(), rnd)
+            try {
+                res.add((encounterProps[encIndex].first))
+            } catch (e: Exception) {
+                throw Exception("Unknown encounter Modifier: ${encounterProps[encIndex]}. Possibilities")
+            }
```

Comment:
why catch an exception and then immedeately rethrow another? It's often better to let the original exception through, in order to preserve the stack trace